src/neo-vm/Types/ByteString.cs

```diff
@@ -63,14 +63,14 @@ internal override bool Equals(StackItem? other, ExecutionEngineLimits limits)
 
         internal bool Equals(StackItem? other, ref uint limits)
         {
+            if (ReferenceEquals(this, other)) return true;
```

Comment:
This should be reverted too. Because we need to execute the `finally` statement.